### DISCUSSION TITLE
Defining Multiline Tasks Through Presets with mise config set

### DISCUSSION THREAD

**Discussion by gabalicious** [jdx/mise#3811] [Ideas]
Here’s your markdown again:

# Discovery: Using `mise config set` for Multiline Tasks

Hi Mise team,

First, I want to say how much I love Mise and how it's been a game-changer for my workflow. Recently, I had an interesting discovery while trying to create a **multiline task**, and I thought it might help others if I shared it here.

## The Problem
I was trying to add a task through a preset with multiline commands. Initially, I attempted to use `mise task add`, but I couldn’t get it to work. No matter how I formatted the commands, they wouldn’t be interpreted correctly or executed in the proper syntax.

## The "Hail Mary" Attempt
After multiple tries—and almost giving up—I decided to rely on AI tools (like Copilot and Anthropic's engine) for suggestions. That’s when it suggested a completely different approach, one I hadn’t seen in the documentation:

```bash
mise config set tasks.setup.run "
python -m venv .venv
source .venv/bin/activate
uv pip install -r requirements.txt
" -t string
```

I thought, “Why not give it a shot?” and it worked! 🎉

The result in the TOML file looked like this:

```
[tasks.setup]
run = """
python -m venv .venv
source .venv/bin/activate
uv pip install -r requirements.txt
"""
```

## What I Learned
- Using `mise config set` with the `-t string` option allowed me to define the multiline commands, preserving the line breaks as needed.
- The `-t` option (e.g., `-t string`, `-t bool`, or other types) might also work in other contexts, which could be worth exploring.

## Suggestion
Could this be added to the documentation? A small section about multiline tasks and the `mise config set` approach  (or `mise task add` if there is a way) would help others who face the same issue. It's especially useful to add tasks as part of a preset.

Thank you for building such an amazing tool, and I hope this discovery can help others as well. If this behavior is intended or there’s a better way to handle multiline tasks, I’d love to learn more!

**Comment by Marukome0743** [comment]
I rechecked this against the current main branch, and the approach described here still works.

For example:

```sh
mise config set tasks.setup.run \
  $'python -m venv .venv\nsource .venv/bin/activate\nuv pip install -r requirements.txt' \
  -t string
```

produces a multiline TOML task equivalent to:

```toml
[tasks.setup]
run = """
python -m venv .venv
source .venv/bin/activate
uv pip install -r requirements.txt"""
```

The line breaks are preserved and the resulting task configuration is valid. Therefore, `mise config set ... -t string` remains a useful way for scripts or presets to create multiline task definitions programmatically.

For manually authored configuration, a multiline `run` string can also be written directly in `mise.toml`.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
